src/libraries/Microsoft.Extensions.Caching.Memory/src/CacheEntry.CacheEntryState.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+
+namespace Microsoft.Extensions.Caching.Memory
+{
+    internal partial class CacheEntry
+    {
+        // this type exists just to reduce CacheEntry size by replacing many enum & boolean fields with one of a size of Int32
+        [StructLayout(LayoutKind.Explicit)]
+        private struct CacheEntryState
+        {
+            // this field overlaps with the 4 other byte fields, it exists to allow for atomic updates
+            [FieldOffset(0)]
+            private int _state;
+
+            [FieldOffset(0)]
+            private byte _flags;
+            [FieldOffset(1)]
+            private byte _evictionReason;
+            [FieldOffset(2)]
+            private byte _priority;
+            [FieldOffset(3)]
+            private byte _reserved; // for future use
+
+            internal CacheEntryState(CacheItemPriority priority) : this() => _priority = (byte)priority;
+
+            internal bool IsDisposed { get => ((Flags)_flags).HasFlag(Flags.IsDisposed); set => SetFlag(Flags.IsDisposed, value); }
+
+            internal bool IsExpired
+            {
+                [MethodImpl(MethodImplOptions.AggressiveInlining)] // added based on profiling
+                get => ((Flags)_flags).HasFlag(Flags.IsExpired);
```

Comment:
```suggestion
                get => ((Flags)_flags & Flags.IsExpired) != 0;
```
This will inline nicely on its own, and avoids a ton of busy work in the JIT to get rid of the boxing and method call that `Enum.HasFlag` compiles into.

Reply:
Same for all other HasFlag uses in this file.